src/storage/leveldb/store.ts

```diff
@@ -356,11 +362,48 @@ export default class LevelDBStore implements IStore {
       await this.addressIndex.clear();
       await this.walletIndex.cleanWalletData();
     }
+    /* Clean registered nano contracts when cleaning tokens */
+    if (cleanTokens) {
+      await this.nanoContractIndex.clear();
+    }
   }
 
   async cleanMetadata(): Promise<void> {
     await this.tokenIndex.clearMeta();
     await this.addressIndex.clearMeta();
     await this.utxoIndex.clear();
   }
+
+  /**
+   * Return if the nano contract is registered for the given address based on ncKey.
+   *
+   * @param ncKey Pair address:ncId concatenated.
+   * @returns `true` if registered and `false` otherwise.
+   * @async
+   */
+  async isNanoContractRegistered(ncKey: string): Promise<boolean> {
+    return this.nanoContractIndex.isNanoContractRegistered(ncKey);
+  }
+
+  /**
+   * Get a nano contract data on storage from the ncKey.
+   *
+   * @param ncKey Pair address:ncId registered.
+   * @returns Nano contract data instance.
+   * @async
+   */
+  async getNanoContract(ncKey: string): Promise<INcData | null> {
+    return this.nanoContractIndex.getNanoContract(ncKey) || null;
```

Comment:
`this.nanoContractIndex.getNanoContract(ncKey)` will return a `Promise<INcData>`, this makes the `|| null` useless.
Either the  `|| null` should be done inside `nanoContractIndex.getNanoContract` or we should await the result then make `|| null` from it

Reply:
Done at https://github.com/HathorNetwork/hathor-wallet-lib/pull/614/commits/ffb0c18da2c771fa8ca9de62538a7f30e8fdd56a